crypto/merkle/hash.go

```diff
@@ -12,33 +10,26 @@ var (
 	innerPrefix = []byte{1}
 )
 
-// returns tmhash(<empty>).
-func emptyHash() []byte {
-	return tmhash.Sum([]byte{})
-}
-
-// returns tmhash(0x00 || leaf).
-func leafHash(leaf []byte) []byte {
-	return tmhash.Sum(append(leafPrefix, leaf...))
-}
-
-// returns tmhash(0x00 || leaf).
-func leafHashOpt(s hash.Hash, leaf []byte) []byte {
-	s.Reset()
-	s.Write(leafPrefix)
-	s.Write(leaf)
-	return s.Sum(nil)
+// returns h(<empty>).
+func emptyHash(h hash.Hash) []byte {
+	h.Reset()
+	h.Write([]byte{})
```

Comment:
Are we sure we need this line?

Reply:
Actually, I checked the `crypto/sha256` implementation and it somehow expects to process empty slices. So please just ignore my comment